### PR TITLE
#88 アップデート時にMySQL DBへぶち込むようにした

### LINKED ISSUE #88: アップロードされたBookmarkをDBにぶち込む

### PR DESCRIPTION
**Pull request by Sorarinu**
Close #88

### CODE REVIEW

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 1]
```diff
@@ -347,6 +352,88 @@ public function upload()
         }
     }
 
+    /**
+     * アップロードデータをMySQLにぶち込む
+     *
+     * @param $data
+     */
+    private function insertDB($data) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Opening brace should be on a new line

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 2]
```diff
@@ -347,6 +352,88 @@ public function upload()
...
+        $user_id = $this->request->session()->get('user_id', function() {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after FUNCTION keyword; 0 found

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 3]
```diff
@@ -347,6 +352,88 @@ public function upload()
...
+                    $bookmark->parent_id = $node['parent_id'];
+                    $bookmark->title = $node['title'];
+                    $bookmark->detail = $node['detail'];
+                    $bookmark->reg_date = $node['reg_date'];
+                    $bookmark->folder = $node['folder'];
+                    $bookmark->url = $node['url'];
+                    $bookmark->save();
+                }
+            }
+        } catch(\Exception $e) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after CATCH keyword; 0 found

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 4]
```diff
@@ -347,6 +352,88 @@ public function upload()
...
+            $this->upload();
+        }
+    }
+
+    /**
+     * MySQLのID + 1を返す
+     *
+     * @return mixed
+     */
+    private function getId() {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Opening brace should be on a new line

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 5]
```diff
@@ -347,6 +352,88 @@ public function upload()
...
+        return DB::table('bookmark')->max('id') + 1;
+    }
+
+    /**
+     * DBにすでにアップロードしてるか判定
+     *
+     * @param $user_id
+     * @return bool
+     */
+    private function checkExists($user_id) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Opening brace should be on a new line

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 6]
```diff
@@ -347,6 +352,88 @@ public function upload()
...
+        }
+    }
+
+    /**
+     * アップロード時にDBから古いデータを消す
+     *
+     * @param $user_id
+     * @return mixed
+     */
+    private function deleteOldData($user_id) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Opening brace should be on a new line

**Comment by Sorarinu** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/gh/63405502/issues/pull_request/93/diff/e7b991b3fc70218cca6275ccee803ff28fede90f?open_modal=1#code_sniffer)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 7]
```diff
@@ -347,6 +352,92 @@ public function upload()
...
     }
 
+    /**
+     * アップロードデータをMySQLにぶち込む
+     *
+     * @param $data
+     */
+    private function insertDB($data)
+    {
+        $user_id = $this->request->session()->get('user_id', function() {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after FUNCTION keyword; 0 found

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 8]
```diff
@@ -347,6 +352,92 @@ public function upload()
...
+                    $bookmark->parent_id = $node['parent_id'];
+                    $bookmark->title = $node['title'];
+                    $bookmark->detail = $node['detail'];
+                    $bookmark->reg_date = $node['reg_date'];
+                    $bookmark->folder = $node['folder'];
+                    $bookmark->url = $node['url'];
+                    $bookmark->save();
+                }
+            }
+        } catch(\Exception $e) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after CATCH keyword; 0 found

**Comment by Sorarinu** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/gh/63405502/issues/pull_request/93/diff/989be8157b85e26216abfdf59ac6ebe719458e22?open_modal=1#code_sniffer)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

**Comment by namazu510** [comment]
よさげなのでは

**Pull request merged**